src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    option. Either enables synchronous replication for system spaces when the
+    synchronous queue is claimed, overriding the user-provided 'is_sync' space
+    option, or falls back to the user-provided 'is_sync' space option.
+]])
+
+M['compat.box_error_serialize_verbose'] = format_text([[
+    Set the verbosity of error objects serialization:
+
+    - `new`: serialize the error message together with other potentially
+    useful fields
+    - `old`: serialize only the error message
+]])
+
+M['compat.box_error_unpack_type_and_code'] = format_text([[
+    Whether to show error fields in `box.error.unpack()`:
+
+    - `new`: do not show `base_type` and `custom_type` fields; do not show the
+    `code` field if it is 0. Note that `base_type` is still accessible for an
+    error object
+    - `old`: show all fields
+]])
+
+M['compat.box_info_cluster_meaning'] = format_text([[
+    Define the behavior of `box.info.cluster`:
+
+    - `new`: show the entire cluster
+    - `old`: show the current replica set
+]])
+
+M['compat.box_session_push_deprecation'] = format_text([[
+    Whether to raise errors on attempts to call the deprecated function
+    `box.session.push`:
+
+    - `new`: raise an error
+    - `old`: do not raise an error
+]])
+
+M['compat.box_space_execute_priv'] = format_text([[
+    Whether the `execute` privilege can be granted on spaces:
+
+    - `new`: an error is raised
+    - `old`: the privilege can be granted with no actual effect
+]])
+
+M['compat.box_space_max'] = format_text([[
+    Set the maximum space identifier (`box.schema.SPACE_MAX`):
+
+    - `new`: 2147483646
+    - `old`: 2147483647
+]])
+
+M['compat.box_tuple_extension'] = format_text([[
+    Controls `IPROTO_FEATURE_CALL_RET_TUPLE_EXTENSION` and
+    `IPROTO_FEATURE_CALL_ARG_TUPLE_EXTENSION` feature bits that define tuple
+    encoding in iproto call and eval requests.
+
+    - `new`: tuples with formats are encoded as `MP_TUPLE`
+    - `old`: tuples with formats are encoded as `MP_ARRAY`
+]])
+
+M['compat.box_tuple_new_vararg'] = format_text([[
+    Controls how `box.tuple.new` interprets an argument list:
+
+    - `new`: as a value with a tuple format
+    - `old`: as an array of tuple fields
+]])
+
+M['compat.c_func_iproto_multireturn'] = format_text([[
+    Controls wrapping of multiple results of a stored C function when returning
+    them via iproto:
+
+    - `new`: return without wrapping (consistently with a local call via
+    `box.func`)
+    - `old`: wrap results into a MessagePack array
+]])
+
+M['compat.console_session_scope_vars'] = format_text([[
+    Whether a console session has its own variable scope.
+
+    In the old behavior, all the non-local variable assignments from the console
+    are written to globals. In the new behavior, they're written to a variable
+    scope attached to the console session.
+
+    Also, a couple of built-in modules are added into the initial console
+    variable scope in the new behavior.
```

Comment:
I like how you described the behavior behind the other options:

```markdown
- `new`: <...>
- `old`: <...>
```

Can we do it here?

Reply:
Done.
```lua
I['compat.console_session_scope_vars'] = format_text([[
    Whether a console session has its own variable scope:

    - `new` (4.x default): all non-local variable assignments
      from the console are written to globals.
    - `old` (3.x default): non-local variable assignments
      are written to a variable scope attached to the console session.
]])
```

Reply:
It seems, new and old behavior are opposite to their descriptions. I mean, `old` describes how `new` works and vice versa.

Reply:
swapped.